### PR TITLE
TPC: change track params in dEdx calibration

### PR DESCRIPTION
**Pull request by badarots**
Correct for Tgl and Snp, insted of Z and Tgl

Other  changes:
* Save run number as CCDB metadata
* ajustable Tgl limits
* record number of tracks of each correction fit
* Method to convert a boost hist into a THn

### CODE REVIEW

File: Detectors/TPC/calibration/include/TPCCalibration/CalibdEdx.h [hunk 1]
```diff
@@ -63,18 +61,18 @@ class CalibdEdx
...
+    FloatAxis, // Snp
+    // FloatAxis, // Z
     IntAxis, // sector
     IntAxis, // stackType
     IntAxis  // Charge
     >;
 
   using Hist = boost::histogram::histogram<AxesType>;
-  using FitCuts = std::array<int, 3>;
+  using FitCuts = std::array<float, 3>;
```
Comment (wiechula) [inline Detectors/TPC/calibration/include/TPCCalibration/CalibdEdx.h]: Why did you change this to float?

Comment (badarots) [reply Detectors/TPC/calibration/include/TPCCalibration/CalibdEdx.h]: For stupid reasons, never mind. Its back an int.

File: Detectors/TPC/calibration/include/TPCCalibration/CalibdEdx.h [hunk 2]
```diff
@@ -130,15 +118,17 @@ class CalibdEdx
   /// Save the histograms to a TTree.
   void writeTTree(std::string_view fileName) const;
 
- private:
-  constexpr static float mipScale = 1.0 / 50.0; ///< Target value for MIP dE/dx
+  constexpr static float mipScale = 1.0 / 50.0; ///< Inverse of target MIP dE/dx value
```
Comment (wiechula) [inline Detectors/TPC/calibration/include/TPCCalibration/CalibdEdx.h:121]:
```suggestion
  constexpr static float MipScale = 1.0 / 50.0; ///< Inverse of target MIP dE/dx value
```
see https://rawgit.com/AliceO2Group/CodingGuidelines/master/naming_formatting.html?showone=Variable_Names#Variable_Names

File: Detectors/TPC/calibration/src/CalibdEdx.cxx [hunk 3]
```diff
@@ -150,9 +167,9 @@ void fitHist(const Hist& hist, CalibdEdxCorrection& corr, TLinearFitter& fitter,
 
         // constexpr float dEdxResolution = 0.05;
         // const double error = dEdx * dEdxResolution / sqrt(counts);
-        const double error = 1. / sqrt(counts);
+        const double error = 1. / counts;
```
Comment (wiechula) [inline Detectors/TPC/calibration/src/CalibdEdx.cxx:170]: I'm still puzzled about what the correct error is ...

**Review by wiechula — CHANGES_REQUESTED** [review]
